pipeline/runtime/run.go

```diff
@@ -110,12 +117,48 @@ func executeRunStep(ctx context.Context, f RunFunc, r *api.StartStepRequest, out
 					})
 				}
 			}
-			return exited, outputs, exportEnvs, artifact, outputsV2, string(optimizationState), err
 		} else if len(r.OutputVars) > 0 {
 			// only return err when output vars are expected
-			return exited, outputs, exportEnvs, artifact, nil, string(optimizationState), err
+			finalErr = err
+			for _, key := range r.OutputVars {
+				if _, ok := outputs[key]; ok {
+					output := &api.OutputV2{
+						Key:   key,
+						Value: outputs[key],
+						Type:  outputVariableTypeString,
+					}
+					outputsV2 = append(outputsV2, output)
+				}
+			}
```

Comment:
Why do we need to add this ?
OutputVars is deprecated so we can skip this

Reply:
pipelines with older yaml can still use this OutputVars fields right ? added this code for backward compatibility

Reply:
I think you can remove this elseif cond altogether